internal/state/validation.go

```diff
@@ -119,6 +119,15 @@ func validateBlock(state State, block *types.Block) error {
 				state.LastBlockTime,
 			)
 		}
+		if !state.ConsensusParams.PBTS.PBTSEnabled(block.Height) && !isBlockSync {
+			medianTime := MedianTime(block.LastCommit, state.LastValidators)
+			if !block.Time.Equal(medianTime) {
+				return fmt.Errorf("invalid block time. Expected %v, got %v",
+					medianTime,
+					block.Time,
+				)
+			}
+		}
 
 	case block.Height == state.InitialHeight:
```

Comment:
Not sure about this, but I think one of the PRs previously forward-ported was removing the requirement that that the timestamp of the first block has to match the genesis time. If that's that case, this condition should _only_ be checked if PBTS is not enabled